app/src/main/java/org/groundplatform/android/usecases/survey/ListAvailableSurveysUseCase.kt

```diff
@@ -57,8 +57,15 @@ constructor(
   private suspend fun getRemoteSurveyList(): Flow<List<SurveyListItem>> {
     val user = userRepository.getAuthenticatedUser()
     val remoteSurveyListFlow = remoteDataStore.getSurveyList(user)
-    return remoteSurveyListFlow.combine(getLocalSurveyList()) { remoteSurveys, localSurveys ->
-      remoteSurveys.map { remoteSurvey -> addOfflineStatus(remoteSurvey, localSurveys) }
+    val remotePublicSurveyListFlow = remoteDataStore.getPublicSurveyList()
+    val localSurveyListFlow = getLocalSurveyList()
+
+    return combine(remoteSurveyListFlow, remotePublicSurveyListFlow, localSurveyListFlow) {
+      remoteSurveys,
+      remotePublicSurveys,
+      localSurveys ->
+      val allRemoteSurveys = remoteSurveys + remotePublicSurveys
```

Comment:
What happens if the survey is available both remotely and via public surveys? Would it be listed twice?

Reply:
fixed